logstreams/src/test/java/io/zeebe/distributedlog/restore/log/LogReplicatorTest.java

```diff
@@ -75,6 +75,22 @@ public void shouldReplicateAgain() {
     assertThat(client.getRequests().get(response.getToPosition())).isNotNull().isNotCompleted();
   }
 
+  @Test
+  public void shouldExcludeFromPositionWhenReplicateAgain() {
+    // given
+    final LogReplicationResponse response = newResponse(true, 2);
+
+    // when
+    final CompletableFuture<Long> result =
+        replicator.replicate(server, -1, response.getToPosition() + 1, true);
+    client.complete(-1, response);
+
+    // then
+    assertThat(client.getRequests()).hasSize(2);
+    assertThat(client.getRequestLog().get(0).includeFromPosition()).isTrue();
```

Comment:
I think we don't need the `requestLog` here.

```java
@Test
public void shouldExcludeFromPositionWhenReplicateAgain() {
  // given
  final LogReplicationResponse response = newResponse(true, 2);

  // when
  final CompletableFuture<Long> result =
      replicator.replicate(server, -1, response.getToPosition() + 1, true);
  client.complete(-1, response);

  // then
  assertThat(client.getRequests()).hasSize(2);
  assertThat(client.getRequests().get(-1).includeFromPosition()).isTrue();
  assertThat(client.getRequests().get(response.getToPosition()).includeFromPosition()).isFalse();
}
```

I think that works?

Reply:
That doesn't work. `client.getRequests().get(response.getToPosition())` returns a 'LogReplicationResponse`. `includeFromPosition` is only included in request now. We could add it also in the response, as an additional check. But I don't think that is necessary.

Reply:
You're right, and no, let's not put it there. Leave it as is then; I think in the future we can simplify all this by having the `replicate` method take in a prepared `LogReplicationRequest` like we did for the restore info stuff, so we could just index by request at that point (and you can look up the keys to get the request). That will also simplify when we change the request, since the SBE stuff can just delegate to an instance of `DefaultLogReplicationRequest`. But we can do that some other time